packages/table/test/table.test.ts

```diff
@@ -0,0 +1,1043 @@
+    });
+
+    it('ensures that virtualized elements with values in .selected are visually selected when brought into view using scrollToIndex', async () => {
+        const virtualItems = makeItemsTwo(50);
+        const el = await fixture<Table>(html`
+            <sp-table
+                selects="multiple"
+                .selected=${['1', '47']}
+                style="height: 120px"
+                .items=${virtualItems}
+                .renderItem=${renderItem}
+                scroller?="true"
+            >
+                <sp-table-head>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                </sp-table-head>
+            </sp-table>
+        `);
+
+        await elementUpdated(el);
+        await oneEvent(el, 'rangeChanged');
+        await elementUpdated(el);
+
+        const rowOne = el.querySelector('[value="1"]') as TableRow;
+        const rowOneCheckboxCell = rowOne.querySelector(
+            'sp-table-checkbox-cell'
+        ) as TableCheckboxCell;
+
+        expect(el.selected).to.deep.equal(['1', '47']);
+        expect(rowOne.selected).to.be.true;
+        expect(rowOneCheckboxCell.checkbox.checked).to.be.true;
+
+        el.scrollToIndex(47);
+
+        await nextFrame();
+        await nextFrame();
+        await elementUpdated(el);
+
+        const unseenRow = el.querySelector('[value="47"]') as TableRow;
+        const unseenRowCheckboxCell = unseenRow.querySelector(
+            'sp-table-checkbox-cell'
+        ) as TableCheckboxCell;
+
+        expect(unseenRow.selected).to.be.true;
+        expect(unseenRowCheckboxCell.checkbox.checked).to.be.true;
+    });
+
+    it('does not set `allSelected` to true by default on Virtualised `<sp-table>`', async () => {
+        const virtualItems = makeItemsTwo(50);
+        const el = await fixture<Table>(html`
+            <sp-table
+                selects="multiple"
+                style="height: 120px"
+                .selected=${['1', '47']}
+                .items=${virtualItems}
+                .renderItem=${renderItem}
+                scroller?="true"
+            >
+                <sp-table-head>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                    <sp-table-head-cell>Column Title</sp-table-head-cell>
+                </sp-table-head>
+            </sp-table>
+        `);
+
+        await elementUpdated(el);
+
+        const tableHeadCheckboxCell = el.querySelector(
+            'sp-table-head sp-table-checkbox-cell'
+        ) as TableCheckboxCell;
+
+        expect(el.selected).to.deep.equal(['1', '47']);
+        expect(tableHeadCheckboxCell.checkbox.checked).to.be.false;
+        expect(tableHeadCheckboxCell.checkbox.indeterminate).to.be.true;
+    });
+});
```

Comment:
This is really great work on the tests here. A little added organization to the suite and it's gonna be a great baseline for future development/feature additions!